src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -2627,7 +2627,7 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             target_type, nullptr, "call_arg_value_ptr");
         builder->CreateStore(llvm_tmp, target);
         llvm_tmp = target;
-        return llvm_tmp;
+        return builder->CreateLoad(llvm_tmp);
```

Comment:
This load is necessary because if you don't do it then the address of `call_arg_value_ptr` is passed and not the address of the variable/array which we are interested in. I am not sure why we needed creation of a variable `call_arg_value_ptr`, putting the address of our variable of interest into it then passing it to the function. Related - https://github.com/lcompilers/lpython/pull/587/files#r895409003

Bottomline - I think `call_arg_value_ptr` is not needed and we can remove the same.

@certik

Reply:
Yes, that's better. Let's implement few more cases and tests. Then we can meet and discuss, I think there must be a way to clean this up.

Reply:
Commented the logic for `call_arg_value_ptr` for now. If needed in future we can uncomment it.